pkg/skaffold/build/gcb/jib.go

```diff
@@ -22,17 +22,22 @@ import (
 	cloudbuild "google.golang.org/api/cloudbuild/v1"
 )
 
-// TODO(dgageot): check that `package` is bound to `jib:build`
-func (b *Builder) jibMavenBuildSteps(artifact *latest.JibMavenArtifact, tag string) []*cloudbuild.BuildStep {
-	return []*cloudbuild.BuildStep{{
+func (b *Builder) jibMavenBuildDescription(artifact *latest.JibMavenArtifact, tag string, base cloudbuild.Build) cloudbuild.Build {
+	base.Steps = []*cloudbuild.BuildStep{{
 		Name: b.MavenImage,
-		Args: jib.GenerateMavenArgs("dockerBuild", tag, artifact, b.skipTests),
+		Args: fixHome(jib.GenerateMavenArgs("build", tag, artifact, b.skipTests)),
 	}}
+	return base
 }
 
-func (b *Builder) jibGradleBuildSteps(artifact *latest.JibGradleArtifact, tag string) []*cloudbuild.BuildStep {
-	return []*cloudbuild.BuildStep{{
+func (b *Builder) jibGradleBuildDescription(artifact *latest.JibGradleArtifact, tag string, base cloudbuild.Build) cloudbuild.Build {
+	base.Steps = []*cloudbuild.BuildStep{{
 		Name: b.GradleImage,
-		Args: jib.GenerateGradleArgs("jibDockerBuild", tag, artifact, b.skipTests),
+		Args: fixHome(jib.GenerateGradleArgs("jib", tag, artifact, b.skipTests)),
 	}}
+	return base
+}
+
+func fixHome(args []string) []string {
+	return append([]string{"-Duser.home=/builder/home"}, args...)
```

Comment:
I am concerned with this workaround, which changes the user home JVM-wise globally affecting all code including Maven and Gradle. For example, the user may be caching `/root/.m2` manually using GCS, and this will be a breaking change. I still believe GCB should fix this at the core. Either set `$HOME` to `/builder/home` or have `.docker/config.json` under `/root`. If GCB can't fix this soon, alternatively I think it is better to introduce a new flag in Jib to allow configuring the location of the Docker config, which can be a useful feature. @GoogleContainerTools/java-tools-build WDYT?

Reply:
I don't see how it's dangerous. The user has no control over how Skaffold uses GCB. So they can't "be caching /root/.m2 manually". Skaffold creates the whole `cloudbuild.yaml` for the user.

Also, GCB does set `HOME` to `/builder/home`.

Reply:
Oh, yeah, my mistake about `$HOME`, sorry. To be fair, I did know `HOME` points to `/builder/home`, which is very confusing actually, because the OS-defined home directory for the user `root` is still `/root`. (The Maven runs as `root` and stores artifacts under `/root/.m2`.)

You may be right that it is not dangerous in the context of Skaffold usage. It is just that I don't really like the idea of switching out the home directory of one user to a different one only because of the issue in GCB, and I think the right solution is that GCB fixes the root cause.

Reply:
Okay, if anyone think this workaround is fine, I am good.

Reply:
Referencing `$HOME` seems safe.  The corresponding bug [JDK-7069190](https://bugs.openjdk.java.net/browse/JDK-7069190) doesn't look likely to be resolved anytime soon.

Reply:
Agreed, "setting home to home" seems reasonable in the Skaffold GCB context.